doc/man3/OSSL_HPKE_enc.pod

```diff
@@ -0,0 +1,346 @@
+=pod
+
+=head1 NAME
+
+OSSL_HPKE_keygen, 
+OSSL_HPKE_suite_check, OSSL_HPKE_str2suite,
+OSSL_HPKE_expansion,OSSL_HPKE_good4grease,
+OSSL_HPKE_CTX_new, OSSL_HPKE_CTX_free,
+OSSL_HPKE_CTX_set1_psk,
+OSSL_HPKE_CTX_set1_ikme,
+OSSL_HPKE_CTX_set1_senderpriv,
+OSSL_HPKE_CTX_set1_authpriv,
+OSSL_HPKE_CTX_set1_authpub,
+OSSL_HPKE_CTX_get0_seq, OSSL_HPKE_CTX_set1_seq,
+OSSL_HPKE_sender_seal, OSSL_HPKE_recipient_open,
+OSSL_HPKE_CTX_export,
+OSSL_HPKE_enc, OSSL_HPKE_dec
+- Hybrid Public Key Encryption (HPKE) functions
+
+=head1 SYNOPSIS
+
+ #include <crypto/hpke.h>
+
+ OSSL_HPKE_CTX *OSSL_HPKE_CTX_new(int mode, OSSL_HPKE_SUITE suite,
+                                  OSSL_LIB_CTX *libctx, const char *propq);
+
+ void OSSL_HPKE_CTX_free(OSSL_HPKE_CTX *ctx);
+
+ int OSSL_HPKE_CTX_set1_psk(OSSL_HPKE_CTX *ctx,
+                            const char *pskid,
+                            const unsigned char *psk, size_t psklen);
+ int OSSL_HPKE_CTX_set1_senderpriv(OSSL_HPKE_CTX *ctx, EVP_PKEY *privp);
+ int OSSL_HPKE_CTX_set1_authpriv(OSSL_HPKE_CTX *ctx, EVP_PKEY *privp);
+ int OSSL_HPKE_CTX_set1_authpub(OSSL_HPKE_CTX *ctx,
+                                unsigned char *pub,
+                                size_t publen);
+ int OSSL_HPKE_CTX_get0_seq(OSSL_HPKE_CTX *ctx, uint64_t *seq);
+ int OSSL_HPKE_CTX_set1_seq(OSSL_HPKE_CTX *ctx, uint64_t seq);
+ int OSSL_HPKE_CTX_set1_ikme(OSSL_HPKE_CTX *ctx,
+                            const unsigned char *ikme, size_t ikmelen);
+
+ int OSSL_HPKE_sender_seal(OSSL_HPKE_CTX *ctx,
```

Comment:
I still wonder if this would be better as 2 calls... This would solve the export only issue also..

Reply:
The seal really only requires the ct, aad and pt on each call..

Reply:
Not for ECH - the AAD changes each time and the INFO input changes based on the specific HTTPS RR/ECHConfig being used. Symmetry with the _open call for ECH in web servers I think also argues for these inputs.

Reply:
This is the RFC defn.... (There are 2 bits , a setup and a seal call).
````
def Seal<MODE>(pkR, info, aad, pt, ...):
  enc, ctx = Setup<MODE>S(pkR, info, ...)
  ct = ctx.Seal(aad, pt)
return enc, ct
````
I am failing to see why this would not work..

Reply:
ended up more or less there